### PR TITLE
convert h264 to mp4

### PR DESCRIPTION
**Pull request by kevin-verschaeve**
it works but it is veeeeeeery slow.

Should try on the rpi, and if too slow, we will need to find something else

### CODE REVIEW

**Comment by Jean-Beru** [comment]
There is nothing to install in Dockerfile ?

**Comment by kevin-verschaeve** [comment]
There already is ffmpeg, see https://github.com/Jean-Beru/baby-foot-replay/blob/develop/.docker/Dockerfile#L3

but, this install the 3.4.4 version that is not the last, maybe, trying with the 4.0 can be faster

File: index.js [hunk 1]
```diff
@@ -54,12 +58,24 @@ app.get('/replay/:last', function(req, res) {
...
+        video
+          .setVideoFormat('mp4')
+          .save(PUBLIC_FOLDER + '/' + finalFile, function (error, file) {
+            if (!error) {
+              fs.unlinkSync(PUBLIC_FOLDER + '/' + tempFile);
+              res.send(finalFile);
+            }
+          });
+      }, function (err) {
+          console.log('Error: ' + err);
```
Comment (Jean-Beru) [inline index.js]: You should also add a `res.statusCode(500).send({ error: err.merssage })`.

File: index.js [hunk 2]
```diff
@@ -54,12 +58,24 @@ app.get('/replay/:last', function(req, res) {
...
+      });
+    }).catch(function() {
+      console.log(arguments);
```
Comment (Jean-Beru) [inline index.js]: Same here.

File: index.js [hunk 3]
```diff
@@ -54,12 +58,24 @@ app.get('/replay/:last', function(req, res) {
...
     .concat()
     .then(function() {
-      res.send(filename);
-    })
-  ;
+      new ffmpeg(PUBLIC_FOLDER + '/' + tempFile).then(function (video) {
+        video
+          .setVideoFormat('mp4')
+          .save(PUBLIC_FOLDER + '/' + finalFile, function (error, file) {
+            if (!error) {
```
Comment (Jean-Beru) [inline index.js]: Early return with `if (error) { res.statusCode(500).send({ error: error.merssage }); return; }`

**Review by Jean-Beru — CHANGES_REQUESTED** [review]

**Comment by Jean-Beru** [comment]
Correct ! You should test this branch on the rpi to see that (and pray).

File: index.js (hunk 3 above)
Comment (Jean-Beru) [reply index.js]: Or better (IMO) :
```js
  videoStitch
    .concat({
      silent: !DEBUG,
      overwrite: true,
    })
    .clips(
      videos
        .slice(videos.length - last)
        .map(function(file) {
          return { fileName: `${PUBLIC_FOLDER + LIVE_FOLDER}/${file}` };
        })
    )
    .output(PUBLIC_FOLDER + '/' + tempFile)
    .concat()
    .then(function() {
      return ffmpeg(PUBLIC_FOLDER + '/' + tempFile);
    })
    .then(function(video) {
      video
        .setVideoFormat('mp4')
        .save(PUBLIC_FOLDER + '/' + finalFile, function (error, file) {
          if (error) {
            throw error;
          }

          fs.unlinkSync(PUBLIC_FOLDER + '/' + tempFile);

          res.send(finalFile);
        });
      })
    .catch(function handleError(error) {
      console.log(error);

      res.status(500).send({ error });
    });
```

**Comment by kevin-verschaeve** [comment]
pr rebased and squashed
useless code removed

do we stay like this or is there anything else to change ?

also, MP4Box is not in the docker image, so this won't work on dev env, should we download and compile it on the Dockerfile ?

**Comment by Jean-Beru** [comment]
I OK for your changes. Maybe we may factorize convert method (MP4Box or ffmpeg) in a separated module (like `stream/index.js`) but it can be done later.
IMO, we should add MP4Box in Dockerfile. You'll have to use classic image instead of alpine because gpac apk is not available...

**Review by Jean-Beru — APPROVED** [review]

**Pull request merged**